Program.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace PasswordChecker
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            PasswordRules passwordRules = new PasswordRules();
+            bool isValid = false;
+            string pass = string.Empty;
+
+            while (!isValid)
+            {
+                Console.WriteLine("Input password to check: ");
+
+                pass = Console.ReadLine();
+
+                isValid = passwordRules.CallTheAllCheckFunctions(pass);
+            }
+            
+            if (isValid)
+            {
+                passwordRules.CheckPasswordByApi(pass);
+            }

```

Comment:
Considering the fact that we have the `while ` loop here, the `if (isValid)` will always evaluate to `true`

I believe you wanted to do something like this:

```suggestion
            while (!isValid)
            {
                Console.WriteLine("Input password to check: ");
                pass = Console.ReadLine();

                isValid = passwordRules.CallTheAllCheckFunctions(pass);
                
                if (isValid) 
                {
                      isValid = PasswordRules.CheckPasswordByApi(pass); //with capital P, as we can have it as a static class 
                }
            }
```